pcb/rules/rule7_4.py

```diff
@@ -0,0 +1,159 @@
+            self.addMessage("Refdes text size ({x}mm) is outside allowed range [{y}mm - {z}mm]".format(
+                x = fh,
+                y = KLC_TEXT_SIZE_MIN,
+                z = KLC_TEXT_SIZE_MAX))
+            err = True
+                
+        # Font thickness
+        if ft < KLC_TEXT_THICKNESS_MIN or ft > KLC_TEXT_THICKNESS_MAX:
+            self.addMessage("Refdes text thickness ({x}mm) is outside allowed range [{y}mm - {z}mm]".format(
+                x = ft,
+                y = KLC_TEXT_SIZE_MIN,
+                z = KLC_TEXT_SIZE_MAX))
+            err = True
+            
+        return err
+    
+    # Check fab line widths
+    def checkIncorrectWidth(self):
+        self.bad_fabrication_width = []
+        for graph in (self.f_fabrication_all + self.b_fabrication_all):
+            if graph['width'] < KLC_FAB_WIDTH_MIN or graph['width'] > KLC_FAB_WIDTH_MAX:
+                self.bad_fabrication_width.append(graph)
+                
+        msg = False
+    
+        if len(self.bad_fabrication_width) > 0:
+            self.addMessage("Some fabrication layer lines have a width outside allowed range of [{x}mm - {y}mm]".format(
+                x = KLC_FAB_WIDTH_MIN,
+                y = KLC_FAB_WIDTH_MAX))
+        for g in self.bad_fabrication_width:
+            self.addMessage("\t- {line}".format(line=g))
+        
+        return len(self.bad_fabrication_width) > 0
+        
+    def check(self):
+        """
+        Proceeds the checking of the rule.
+        The following variables will be accessible after checking:
+            * f_fabrication_all
+            * b_fabrication_all
+            * f_fabrication_lines
+            * b_fabrication_lines
+            * bad_fabrication_width
+        """
+        module = self.module
+        self.f_fabrication_all = module.filterGraphs('F.Fab')
+        self.b_fabrication_all = module.filterGraphs('B.Fab')
+
+        self.f_fabrication_lines = module.filterLines('F.Fab')
+        self.b_fabrication_lines = module.filterLines('B.Fab')
+                
+        return any([
+                    self.checkMissingValue(),
+                    self.checkMissingLines(),
+                    self.checkIncorrectWidth(),
+                    self.checkSecondRef(),
+                    ])
+
+    def fix(self):
+        """
+        Proceeds the fixing of the rule, if possible.
+        """
+        module = self.module
+        if self.checkIncorrectWidth():
+            for graph in self.bad_fabrication_width:
+                graph['width'] = KLC_FAB_WIDTH
+                
+        if self.checkMissingValue():
+            module.value['layer'] = 'F.Fab'
+            module.value['font']['height'] = self.expected_val_width
+            module.value['font']['width'] = self.expected_val_width
+            module.value['font']['thickness'] = self.expected_val_thickness
+            
+        if self.checkSecondRef():
+            pass
```

Comment:
Why did you delete the creation of a second %R-label? Code was present in `pcb/rules/rule10_5.py`

```python
if module.needs_secondrefdes:
                txt={'user': '%R', 'pos': module.secondrefdespos, 'layer': module.secondrefdeslayer, 'font': module.secondrefdesfont, 'hide': False } 
                module.userText.append(txt)
                print('added second REFDES')
            else:
                for txt in range(0,len(module.userText)):
                    if module.userText[txt]['user']=='%R':
                        print('OLD: ', module.userText[txt])
                        module.userText[txt]['pos']=module.secondrefdespos
                        module.userText[txt]['layer']=module.secondrefdeslayer
                        module.userText[txt]['font']=module.secondrefdesfont
                        print('NEW: ', module.userText[txt])
                        print('fixed second REFDES ', txt)
                        break

```